src/Microsoft.DotNet.MSBuildSdkResolver/MSBuildSdkResolver.cs

```diff
@@ -292,17 +308,26 @@ private NETCoreSdkResolver.Result ResolveNETCoreSdkDirectory(SdkResolverContext
         {
             string globalJsonStartDir = Path.GetDirectoryName(context.SolutionFilePath ?? context.ProjectFilePath);
             var result = NETCoreSdkResolver.ResolveSdk(dotnetExeDir, globalJsonStartDir, _vsSettings.DisallowPrerelease());
+            result.GlobalJsonHonored = true;
```

Comment:
nit: it is better not to mutate states after it is set. Use a non initialized local variable, and then set result.GlobalJsonHonored to it after the branch logic is done.